src/libs/actions/Transaction.ts

```diff
@@ -719,14 +719,15 @@ function changeTransactionsReport(
     const failureData: OnyxUpdate[] = [];
     const successData: OnyxUpdate[] = [];
 
-    const existingSelfDMReportID = findSelfDMReportID();
-    let selfDMReport: Report | undefined;
+    let selfDMReportID = findSelfDMReportID();
+    let selfDMReport: Report | undefined = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${selfDMReportID}`];
```

Comment:
Not sure if this method is pure, but we should pass the selfDM report in here to make it pure

Reply:
I think you would need to pass the report, not just the id to make it pure, but lets not do it here